x-pack/plugins/lens/public/editor_frame_service/editor_frame/workspace_panel/chart_switch.tsx

```diff
@@ -294,24 +296,46 @@ export const ChartSwitch = memo(function ChartSwitch(props: Props) {
-                    append:
-                      v.selection.dataLoss !== 'nothing' ? (
-                        <EuiIconTip
-                          aria-label={i18n.translate('xpack.lens.chartSwitch.dataLossLabel', {
-                            defaultMessage: 'Warning',
-                          })}
-                          type="alert"
-                          color="warning"
-                          content={i18n.translate('xpack.lens.chartSwitch.dataLossDescription', {
-                            defaultMessage:
-                              'Selecting this chart type will result in a partial loss of currently applied configuration selections.',
-                          })}
-                          iconProps={{
-                            className: 'lnsChartSwitch__chartIcon',
-                            'data-test-subj': `lnsChartSwitchPopoverAlert_${v.id}`,
-                          }}
-                        />
-                      ) : null,
+                    append: (
+                      <EuiFlexGroup
+                        gutterSize="xs"
+                        responsive={false}
+                        style={{ display: 'inline-flex' }}
+                      >
+                        {v.selection.dataLoss !== 'nothing' ? (
+                          <EuiFlexItem grow={false}>
+                            <EuiIconTip
+                              aria-label={i18n.translate('xpack.lens.chartSwitch.dataLossLabel', {
+                                defaultMessage: 'Warning',
+                              })}
+                              type="alert"
+                              color="warning"
+                              content={i18n.translate(
+                                'xpack.lens.chartSwitch.dataLossDescription',
+                                {
+                                  defaultMessage:
+                                    'Selecting this chart type will result in a partial loss of currently applied configuration selections.',
+                                }
+                              )}
+                              iconProps={{
+                                className: 'lnsChartSwitch__chartIcon',
+                                'data-test-subj': `lnsChartSwitchPopoverAlert_${v.id}`,
+                              }}
+                            />
+                          </EuiFlexItem>
+                        ) : null}
+                        {v.showBetaBadge ? (
+                          <EuiFlexItem grow={false}>
+                            <EuiBadge color="hollow">
+                              <FormattedMessage
+                                id="xpack.lens.chartSwitch.betaLabel"
+                                defaultMessage="Beta"
+                              />
+                            </EuiBadge>
+                          </EuiFlexItem>
+                        ) : null}
+                      </EuiFlexGroup>
```

Comment:
I think this `EuiFlexGroup` needs to be made conditional, otherwise an empty `EuiFlexGroup` is rendered when `v.selection.dataLoss === 'nothing'` and `!v.showBetaBadge`.

Reply:
wrapped `EuiFlexGrop` with a condition in https://github.com/elastic/kibana/pull/97978/commits/6d35fb72f453d08422c6e27b43236816d91ba0ee